### ISSUE TITLE
Error by saving of the map

### ISSUE THREAD

**Issue by Guenda85** [iobroker-community-adapters/ioBroker.mihome-vacuum#735]
**Describe the bug**  
Map is not saved

**To Reproduce**  
start download of map by using "mihome-vacuum.0.cleanmap.loadMap"

**Expected behavior**  
successful download of the map

**Screenshots & Logfiles**  
	Line 187378: 2023-01-09 20:55:47.128  - [32minfo[39m: mihome-vacuum.0 (906552) Plugin sentry Sentry Plugin disabled for this process because sending of statistic data is disabled for the system
	Line 187379: 2023-01-09 20:55:47.224  - [32minfo[39m: mihome-vacuum.0 (906552) starting. Version 3.9.2 in /opt/iobroker/node_modules/iobroker.mihome-vacuum, node: v14.20.1, js-controller: 4.0.24
	Line 187379: 2023-01-09 20:55:47.224  - [32minfo[39m: mihome-vacuum.0 (906552) starting. Version 3.9.2 in /opt/iobroker/node_modules/iobroker.mihome-vacuum, node: v14.20.1, js-controller: 4.0.24
	Line 187380: 2023-01-09 20:55:47.358  - [32minfo[39m: mihome-vacuum.0 (906552) IOT enabled, create state
	Line 187384: 2023-01-09 20:55:47.477  - [32minfo[39m: mihome-vacuum.0 (906552) select standard vacuum protocol....
	Line 187384: 2023-01-09 20:55:47.477  - [32minfo[39m: mihome-vacuum.0 (906552) select standard vacuum protocol....
	Line 187385: 2023-01-09 20:55:47.994  - [32minfo[39m: mihome-vacuum.0 (906552) settest next timer: not available
	Line 187388: 2023-01-09 20:55:48.341  - [32minfo[39m: mihome-vacuum.0 (906552) create states for mop
	Line 187389: 2023-01-09 20:55:48.349  - [32minfo[39m: mihome-vacuum.0 (906552) change states from State control.fan_power
	Line 187390: 2023-01-09 20:55:48.350  - [32minfo[39m: mihome-vacuum.0 (906552) create states for water box
	Line 187393: 2023-01-09 20:55:49.379  - [32minfo[39m: mihome-vacuum.0 (906552) create state for carpet_mode
	Line 187669: 2023-01-09 20:58:00.510  - [31merror[39m: mihome-vacuum.0 (906552) Cannot write file actualMap_0.png: mihome-vacuum.admin is not an object of type "meta"
	Line 187669: 2023-01-09 20:58:00.510  - [31merror[39m: mihome-vacuum.0 (906552) Cannot write file actualMap_0.png: mihome-vacuum.admin is not an object of type "meta"
	Line 187670: 2023-01-09 20:58:00.511  - [31merror[39m: mihome-vacuum.0 (906552) Error by saving of the map
	Line 188894: 2023-01-09 21:06:12.537  - [31merror[39m: mihome-vacuum.0 (906552) Cannot write file actualMap_0.png: mihome-vacuum.admin is not an object of type "meta"
	Line 188894: 2023-01-09 21:06:12.537  - [31merror[39m: mihome-vacuum.0 (906552) Cannot write file actualMap_0.png: mihome-vacuum.admin is not an object of type "meta"
	Line 188895: 2023-01-09 21:06:12.538  - [31merror[39m: mihome-vacuum.0 (906552) Error by saving of the map
![2023-01-09 20_58_05-logs - GKE-SV-Linux - Geschäftlich – Microsoft​ Edge](https://user-images.githubusercontent.com/45709724/211398851-2a717533-4c4e-4d44-801b-866e22384f9d.png)
![2023-01-09 20_58_19-objects - GKE-SV-Linux - Geschäftlich – Microsoft​ Edge](https://user-images.githubusercontent.com/45709724/211398876-f029ab08-8582-4939-9759-99a09d37609b.png)

**Versions:**

-   Adapter version: 3.9.2
-   JS-Controller version: 4.0.24 
-   Node version: 14.20.1 
-   vacuum cleaner: roborock.vacuum.s6

**Comment by dirkhe** [comment]
can you check, if iobroker has rights to write to /iobroker/data/iobroker-data/files/mihome-vacuum.admin/actualMap_0.png ? What I'm wondering is the type meta? Did you you use a non FS? Didi you install the webs adapter?

**Comment by Apollon77** [comment]
@dirkhe Question: Why the fiel is stored in .admin??

**Comment by dirkhe** [comment]
Didn't ask me, I 've only checked the code. I think a better option will be /iobroker/data/iobroker-data/files/mihome-vacuum/actualMap_0.png. But I guess, if I changed this, a lot of peaople start crying... The Url is stored in a DP, but you know, copy and paste is easy :)

**Comment by Guenda85** [comment]
permissions seem to be ok:
![image](https://user-images.githubusercontent.com/45709724/211601941-e594bc69-8b32-4eab-ab1d-b6196028085e.png)

iobroker is installed on a default Ubuntu installation. If you're meaning "web" adapter, yes it is installed. In addition maybe it's worth mentioning that the map funktionality used to work in the past

**Comment by Guenda85** [comment]
of course iobroker has no permission to write to /iobroker/data/iobroker-data/files/mihome-vacuum.admin/actualMap_0.png because it is installed in /opt/iobroker - hopefully you're not using absolute paths :-D

**Comment by dirkhe** [comment]
No, it is 
adapter.writeFile('mihome-vacuum.admin', ....

**Comment by Guenda85** [comment]
just made a test and created structure with permissions for user iobroker - still not working:
![image](https://user-images.githubusercontent.com/45709724/211603589-77f84775-9b80-4e8b-9de3-fbcd16d5dc7d.png)

**Comment by Apollon77** [comment]
NO!!!! Please do not create strange directories! This is wrong and not used. In fact yes it is /opt/iobroker/iobroker-data/files/mihome-vacuum.admin/actualMap_0.png when writeFile ('mihome-vacuum.admin", ...) is used

@dirkhe 
> Re  /iobroker/data/iobroker-data/files/mihome-vacuum/actualMap_0.png.

no also not there because this is not accessble by web or other adapters

The right way would be to create an own meta.user filespace for the instance and write it there .. but that admin location works for now

**Comment by Guenda85** [comment]
was just a test :-) yeah i was wondering because the type "meta" is missing as i already highlighted in my first post (red rectangle)

**Comment by dirkhe** [comment]
Ingo, ich schreib mal auf deutsch. 
Ich glaube, die Url ist so auf jeden Fall nutzber, denn sowit ich weiss, nutzten die das in vis, indem die einfach den DP da reinschreiben und in dem steht dann 
"/mihome-vacuum.admin/actualMap_0.png" -> wenn man das dann in einer vis n utzt, kommt da raus http://iobroker:8082/mihome-vacuum.admin/actualMap_0.png -> und das ist erreichbar.

wenn ich jetzt 
```   {
      "_id": "",
      "type": "meta",
      "common": {
        "name": "User files for mihome-vacuum",
        "type": "meta.user"
      },
      "native": {}
    },
```
anlege,  (id muss leerbleiben?)
würde der code dann so aussehen:
```
						adapter.writeFile(`mihome-vacuum.${adapter.instance}`, `actualMap.png`, buf, error => {
							if (error) {
								adapter.log.error('Error by saving of the map');
							} else {
								adapter.setState('cleanmap.mapURL', `/mihome-vacuum.${adapter.instance}/actualMap.png`, true);
							}
							this.mapLastSave = Date.now();
						});
```
und ich könnte dann so drauf zugreifen:
http://iobroker:8082/mihome-vacuum.0/actualMap.png ???

Die frage ist nur, warum klappt das bei allen andren bisher so?

**Comment by Apollon77** [comment]
Am Ende klappt beides, es ist nur so das "*.admin" nicht für user created content gedacht ist sondern das oben genannte der richtige Weg ist. Und ja der Pfad sollte so passen.

**Comment by dirkhe** [comment]
Funktioniert auf jeden Fall alles nicht, jetzt bekomme ich auch die Meldung mit dem fehlenden Meta Daten, obwohl ich das wie oben geschrieben gesetzt habe.
Habe den Adapter sogar neu angelegt, das gleiche. Ich habe es mal in den master comitted, mache aber noch keine npm Version. Für heute habe ich keinen Bock mehr

**Comment by dirkhe** [comment]
Also funktioniert jetzt, es waren 2 Probleme:
1, nutze ich den dev-server, da muss man extrat ein upload senden, damit die io-package.josn neu eingelesen wird.
2. es gab schon ein  "_id": "",  "type": "meta", -> habe ich jetzt anders gelöst, dass ich einen Datenpunkt "userfiles" als id genommen habe

Sollte also alles in der 3.9.3 gelöst sein
